RecoLocalCalo/HcalRecAlgos/src/HcalMahiPulseOffsetsGPU.cc

```diff
@@ -0,0 +1,36 @@
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalMahiPulseOffsetsGPU.h"
+
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+// FIXME: add proper getters to conditions
+HcalMahiPulseOffsetsGPU::HcalMahiPulseOffsetsGPU(edm::ParameterSet const& ps) 
+{
+    auto const& values = ps.getParameter<std::vector<int>>("pulseOffsets");
+    values_.resize(values.size());
+    std::copy(values.begin(), values.end(), values_.begin());
+}
+
+HcalMahiPulseOffsetsGPU::Product::~Product() {
+  // deallocation
+  cudaCheck(cudaFree(values));
+}
+
+HcalMahiPulseOffsetsGPU::Product const& HcalMahiPulseOffsetsGPU::getProduct(cudaStream_t cudaStream) const {
+  auto const& product =
+      product_.dataForCurrentDeviceAsync(cudaStream, [this](HcalMahiPulseOffsetsGPU::Product& product, cudaStream_t cudaStream) {
+        // malloc
+        cudaCheck(cudaMalloc((void**)&product.values, this->values_.size() * sizeof(int)));
```

Comment:
@makortel is there any reason we shouldn't use the caching allocator here as well ?

Reply:
The current caching allocator API does not suit well for the ESProduct case, because the life time of the memory is 
not related to the processing queued to the argument `cudaStream_t`.

I do have an idea (old prototype if I still manage to find it) on how to improve on that on top of https://github.com/cms-patatrack/cmssw/pull/412, although given https://github.com/cms-patatrack/cmssw/pull/487 I'd do it a bit differently.

Reply:
Maybe I should interpret the silence on the RFC's as "no strong objections" and just go ahead with further prototyping.

Reply:
I a re-read to remind myself about it once 11.1.0-pre2 Patatrack is out...